File: Parallel-Stream-API/src/main/java/com/blog/Sum.java
package com.blog;

import java.util.stream.Stream;

public class Sum {
    /**
     * 순차 리듀싱
     * 무한 스트림을 사용해 합계를 계산한다.
     */
    public long sequentialSum(long n) {
        return Stream.iterate(1L, i -> i + 1)
                .limit(n)
                .parallel()
                .reduce(0L, Long::sum);
    }

    /**
     * 반복형
     * 스트림을 사용하지 않고 전통적인 방식으로 합계를 계산해 반환한다.
     */
    public long iterativeSum(long n) {
        long result = 0;
        for (long i = 1L; i <= n; i++) {
            result += i;
        }

        return result;
    }
}

File: type-interface/sum.ts

interface SumFunc { // sum 함수를 만들기 전 데이터 타입을 정의할 interface 선언
    (a: number, b:number): number; //인자와 리턴값의 데이터 타입을 정의
  }

  const sumNum: SumFunc = function (a: number, b: number) {
    return a + b;
  }
  var result = sumNum(5,10);
  console.log(result); 
  
  
  
  
  
  

  